### ISSUE TITLE
Issue Http URI metrics don't respect query parameters

### ISSUE THREAD

**Issue by pendula95** [vert-x3/vertx-dropwizard-metrics#93]
If I have config like this:

```
{
    "metricsOptions": { 
        "enabled": true,
        "monitoredHttpServerUris": [
            {
                "value" : "/test",
                "type" : "EQUALS"
            }
        ]
    }
}
```

Than only requests that go to EXACTLY test are monitored. For example if I visit test?para1=2 than values are not updated for /test

If I add regex configuration like this:

```
{
    "metricsOptions": { 
        "enabled": true,
        "monitoredHttpServerUris": [
            {
                "value" : "\/test.*",
                "type" : "REGEX"
            }
        ]
    }
}
```
Than for each different combination of params new WHOLE record will be created in metricsand I will have something like this:

```
"vertx.http.servers.0.0.0.0:8888.get-requests./test?" : {
    "type" : "timer", ....
```

```
"vertx.http.servers.0.0.0.0:8888.get-requests./test?para1=2" : {
    "type" : "timer", ....
```

```
"vertx.http.servers.0.0.0.0:8888.get-requests./test?para1=3" : {
    "type" : "timer", ....
```

**Comment by pendula95** [comment]
I think the issue is here as uri is passed together with parameters 
https://github.com/vert-x3/vertx-dropwizard-metrics/blob/71085e8b173e0b368551325678582fa71066750d/src/main/java/io/vertx/ext/dropwizard/impl/HttpServerMetricsImpl.java#L45

**Comment by vietj** [comment]
in practice parameters are part of the request URI from the HTTP request perspective

**Comment by vietj** [comment]
so I think the dropwizard implementation should parse that if possible

**Comment by pendula95** [comment]
Yes path parameters but query parameters I don't think should be part. They are changeable in our case /auth?ip={ip}&port={port}&service-uri={service-uri} which change constantly and this would result in massive json that would be unparsable.

**Comment by pendula95** [comment]
If this is not reasonable there should be an option to alter the behaviour like this.

**Comment by vietj** [comment]
I just mean that in `request.uri()` represent the `requestURI` defined by HTTP specification (`Request-Line   = Method SP Request-URI SP HTTP-Version CRLF`) that is opaque and includes parameters when they are present.

**Comment by pendula95** [comment]
I understand that this is by definition by in practise this will cause a lot of issues. On our test simulation the json got larger than 1GB only after 1h of usage when generating snapshot. Is it possible to add some configuration so path or query params are ignored from uri?

**Comment by pendula95** [comment]
Maybe this should be defined differently. What we are look for is endpoint metrics and what current Dropwizard exposes are url metrics. Could it be better we those 2 are both exposed, url metrics as current implementation and endpoint metrics as suggested by me?
